src/main/java/org/cryptomator/ui/health/CheckListController.java

```diff
@@ -1,119 +1,100 @@
 	private final Stage window;
-	private final ObservableList<HealthCheckTask> tasks;
-	private final FilteredList<HealthCheckTask> chosenTasks;
+	private final ObservableList<Check> checks;
+	private final CheckExecutor checkExecutor;
+	private final FilteredList<Check> chosenChecks;
 	private final ReportWriter reportWriter;
-	private final ExecutorService executorService;
-	private final ObjectProperty<HealthCheckTask> selectedTask;
+	private final ObjectProperty<Check> selectedCheck;
+	private final BooleanBinding mainRunStarted; //TODO: rerunning not considered for now
+	private final BooleanBinding somethingsRunning;
 	private final Lazy<ErrorComponent.Builder> errorComponentBuilder;
-	private final SimpleObjectProperty<Worker<?>> runningTask;
-	private final Binding<Boolean> running;
-	private final Binding<Boolean> finished;
 	private final IntegerBinding chosenTaskCount;
 	private final BooleanBinding anyCheckSelected;
-	private final BooleanProperty showResultScreen;
 
 	/* FXML */
-	public ListView<HealthCheckTask> checksListView;
+	public ListView<Check> checksListView;
 
 	@Inject
-	public CheckListController(@HealthCheckWindow Stage window, Lazy<List<HealthCheckTask>> tasks, ReportWriter reportWriteTask, ObjectProperty<HealthCheckTask> selectedTask, ExecutorService executorService, Lazy<ErrorComponent.Builder> errorComponentBuilder) {
+	public CheckListController(@HealthCheckWindow Stage window, List<Check> checks, CheckExecutor checkExecutor, ReportWriter reportWriteTask, ObjectProperty<Check> selectedCheck, Lazy<ErrorComponent.Builder> errorComponentBuilder) {
 		this.window = window;
-		this.tasks = FXCollections.observableList(tasks.get(), HealthCheckTask::observables);
-		this.chosenTasks = this.tasks.filtered(HealthCheckTask::isChosenForExecution);
+		this.checks = FXCollections.observableList(checks, Check::observables);
+		this.checkExecutor = checkExecutor;
+		this.chosenChecks = this.checks.filtered(Check::isChosenForExecution);
 		this.reportWriter = reportWriteTask;
-		this.executorService = executorService;
-		this.selectedTask = selectedTask;
+		this.selectedCheck = selectedCheck;
 		this.errorComponentBuilder = errorComponentBuilder;
-		this.runningTask = new SimpleObjectProperty<>();
-		this.running = EasyBind.wrapNullable(runningTask).mapObservable(Worker::runningProperty).orElse(false);
-		this.finished = EasyBind.wrapNullable(runningTask).mapObservable(Worker::stateProperty).map(END_STATES::contains).orElse(false);
-		this.chosenTaskCount = Bindings.size(this.chosenTasks);
-		this.anyCheckSelected = selectedTask.isNotNull();
-		this.showResultScreen = new SimpleBooleanProperty(false);
+		this.chosenTaskCount = Bindings.size(this.chosenChecks);
+		this.mainRunStarted = Bindings.isEmpty(this.checks.filtered(c -> c.getState() == Check.CheckState.RUNNABLE));
+		this.somethingsRunning = Bindings.isNotEmpty(this.checks.filtered(c -> c.getState() == Check.CheckState.SCHEDULED || c.getState() == Check.CheckState.RUNNING));
```

Comment:
Maybe add convenience getters:

```suggestion
		this.mainRunStarted = Bindings.isEmpty(this.checks.filtered(Check::isRunnable));
		this.somethingsRunning = Bindings.isNotEmpty(this.checks.filtered(Check::isScheduledOrRunning));
```